R/monthly_downscaling.R

```diff
@@ -129,3 +129,53 @@ monthly_downscaling <- function(frac, fld_data, fld_coordinates, fld_time, var){
   list(data = monthly_data, coordinates = fld_coordinates)
 
 }
+
+
+#' Wrapper function for running downscaling from cassandra component
+#'
+#' Convert inputs received from the cassandra component to the format expected
+#' by \code{\link{monthly_downscaling}} and pass the converted arguments.
+#'
+#' The conversions performed are
+#' \describe{
+#' \item{frac}{Dataset is looked up by name.}
+#' \item{fld_data}{The downscaling function wants a single matrix.  This
+#' function accepts a list of matrices and calls the downscaling function on
+#' each one sequentially.}
+#' \item{coords}{The matrix of coordinate data is converted to a data
+#' frame, and the index column is added.}
+#' \item{time}{No conversion.}
+#' \item{var}{No conversion.}
+#' }
+#'
+#' @param frac Name of monthly fraction dataset to use.
+#' @param fld_data List of matrices of annual field data to downscale to monthly.
+#' @param coords Coordinate matrix for the field data, such as the one
+#' produced by \code{\link[fldgen]{coord_array}}.
+#' @param time Vector of years for the input field matrices.
+#' @param var String indicating the variable being downscaled.  Either 'tas' or
+#' 'pr'.
+#' @return List of matrices of downscaled data.  Grid cells are in rows, months
+#' in columns (This is the opposite of the convention used everywhere else in
+#' this package.)
+#' @export
+#' @keywords internal
+downscaling_component_api <- function(frac, fld_data, coords, time, var)
+{
+    frac <- get(frac)   # would be better to use getFromNamespace, but it doesn't seem to work.
```

Comment:
Should we add a check to to make sure that the data that is being called with get exists as internal data? While I like that when we switch to `frac_ipsl_cm5a_lr` from `ipsl-cm5a-lr_frac` we can get rid of the `` in the code so that we can use `get()` to access the data which is great. My concern is that by moving from the ISIMP/CMIP nomenclature we might run into issues with mismatching names.... So may be we add documentation/a note about the change in nomenclature? Or do you think that won't be a problem? Alternatively we could add mapping file to convert from the info used to train `fldgen` to the fraction files we want to use. Or do we want the `fldgen` outputs to reflect this naming pattern?

Reply:
I don't think this is going to be a problem in practice.  This name gets copied directly from the cassandra configuration file into the first argument of this function, so, basically, the remedy is to check the an2month package documentation for the names of datasets that are valid.

With regard to fldgen, the way the fldgen component currently works is that it expects to load a trained emulator from an RDS file (this is to prevent us endlessly rerunning the training procedure on data that isn't changing), so as long as we use the same naming convention for saving our emulator files that we use in this package, it should be easy enough to keep them straight.